openwisp_monitoring/device/models.py

```diff
@@ -150,12 +150,14 @@ def add_mac_vendor_info(self):
             for client in interface['wireless']['clients']:
                 client['vendor'] = self._mac_lookup(client['mac'])
         for neighbor in self.data.get('neighbors', []):
-            neighbor['vendor'] = self._mac_lookup(neighbor['mac_address'])
+            # Neighbors with state = FAILED have no mac_address
+            mac_address = neighbor.get('mac_address', '')
+            neighbor['vendor'] = self._mac_lookup(mac_address)
```

Comment:
if the mac address is empty I would avoid the lookup altogether because it has some cost from a performance point of view, the table with all the mac address prefixes is large